packages/terra-dropdown-button/src/DropdownButton.jsx

```diff
@@ -57,29 +57,57 @@ class DropdownButton extends React.Component {
     this.handleDropdownRequestClose = this.handleDropdownRequestClose.bind(this);
     this.handleKeyDown = this.handleKeyDown.bind(this);
     this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.setButtonNode = this.setButtonNode.bind(this);
+    this.getButtonNode = this.getButtonNode.bind(this);
+    this.setListNode = this.setListNode.bind(this);
+    this.state = { isOpen: false, isActive: false, isKeyboardEvent: false };
+  }
+
+  setListNode(element) {
+    this.dropdownList = element;
+  }
 
-    this.state = { isOpen: false, isActive: false };
+  setButtonNode(node) {
+    this.buttonNode = node;
+  }
+
+  getButtonNode() {
+    return this.buttonNode;
   }
 
   handleDropdownButtonClick() {
+    if (this.state.isOpen) {
+      this.setState({ isKeyboardEvent: false });
+    }
     this.setState(prevState => ({ isOpen: !prevState.isOpen }));
   }
 
   handleDropdownRequestClose(callback) {
     this.setState({ isOpen: false }, typeof callback === 'function' ? callback : undefined);
+    this.setState({ isKeyboardEvent: false, isActive: false });
   }
 
   /*
     In FireFox active styles don't get applied on space
    */
   handleKeyDown(event) {
-    if (event.keyCode === KeyCode.KEY_SPACE) {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
       this.setState({ isActive: true });
+      this.setState({ isKeyboardEvent: true });
+    } else if (event.keyCode === KeyCode.KEY_DOWN && this.state.isOpen && !this.state.isKeyboardEvent) {
+      // puts focus on first list element on down arrow key press when dropdown is opened by mouse click
+      // the div inside the li is what is actually focusable so need to go 2 layers down
+      this.dropdownList.childNodes[0].childNodes[0].focus();
```

Comment:
This will break if the semantic structure of the list ever changes. It might be better to query the dropdownList for a specific data attribute on the focusable element. Or query for list items and focus the first item in the first list item.

Reply:
Does this introduce or resolve any of the voice over concerns? Specifically swipe gesture on iOS. Do they register as keyboard or mouse interactions? How is focus shifted using voice over on desktop and on mobile?

Reply:
Changed it to [data-att ](https://github.com/cerner/terra-core/blob/cf4a1d2744dbd30b29ba636528a1ebd6c739c3a0/packages/terra-dropdown-button/src/DropdownButton.jsx#L99).

Yes this was added to resolve navigation issue that was found while testing with VO and JAWS. Due to holding focus with button on open was making dropdown list inaccessible for VO and JAWS hence handled this in keydown event.

this issue was happening only with keyboard and not with touch / swipe gestures so this has no impact on them.

I have tested for swipe gesture on IOS it works as expected.